prime-router/src/main/kotlin/azure/ConditionMapper.kt

```diff
@@ -34,16 +42,39 @@ class LookupTableConditionMapper(metadata: Metadata) : IConditionMapper {
                 acc
             }
     }
+
+    override fun lookupMemberOid(codings: List<Coding>): Map<String, String> {
+        // Extract condition codes using the mapping table, not directly from codings
+        val testCodes = codings.mapNotNull { it.code } // These are the input test codes
+
+        // Filter rows related to condition mappings based on test codes
+        val filteredRows = mappingTable.FilterBuilder()
+            .isIn(ObservationMappingConstants.TEST_CODE_KEY, testCodes) // Map test codes to conditions
+            .filter().caseSensitiveDataRowsMap
+
+        // Create a map of condition codes to member OIDs
+        return filteredRows.fold(mutableMapOf()) { acc, condition ->
```

Comment:
nit: 
functional programming tip: You want to avoid mutable variables.
```kotlin
return filteredRows.fold(emptyMap()) { acc, condition ->
// ...
if (conditionCode.isNotEmpty() && memberOid.isNotEmpty()) {
    acc + (conditionCode to memberOid)
} else {
  acc
}
```